model/renderers/nemotron3nano.go

```diff
@@ -234,6 +160,59 @@ func (r *Nemotron3NanoRenderer) renderTools(tools []api.Tool) string {
 	return sb.String()
 }
 
+func (r *Nemotron3NanoRenderer) buildContent(message api.Message) string {
+	if message.Thinking != "" {
+		return "<think>\n" + message.Thinking + "\n</think>\n" + message.Content
+	}
+	content := message.Content
+	if !strings.Contains(content, "<think>") && !strings.Contains(content, "</think>") {
+		return "<think></think>" + content
+	}
+	return content
+}
+
+func (r *Nemotron3NanoRenderer) formatContent(content string, truncate bool, addNewline bool) string {
+	if content == "" {
+		return "<think></think>"
+	}
+
+	if !truncate {
+		if addNewline {
+			return strings.TrimSpace(content) + "\n"
+		}
+		return strings.TrimSpace(content)
+	}
+
+	// Truncate thinking - keep only content after </think>
+	c := content
+	if strings.Contains(c, "</think>") {
+		parts := strings.Split(c, "</think>")
```

Comment:
this looks like the kind of logic that's not needed for us since we have separate thinking fields always (assuming the parser is correct). So we shouldn't need to manually manage it in the renderer